ovn-tester/ovn_workload.py

```diff
@@ -441,9 +478,12 @@ def ping_ports(self, cluster, ports):
 
     def get_connection_string(self, cluster_cfg, port):
         protocol = "ssl" if cluster_cfg.enable_ssl else "tcp"
-        offset = 0
-        offset += 3 if cluster_cfg.clustered_db else 1
-        offset += cluster_cfg.n_relays
+        offset = cluster_cfg.n_az
+        if cluster_cfg.clustered_db:
+            offset *= 3
+        if cluster_cfg.n_relays > 0:
+            offset += cluster_cfg.n_relays * cluster_cfg.n_az
+        offset += self.id
         return f"{protocol}:{self.mgmt_ip + offset}:{port}"
```

Comment:
Same here, I don't think this is exactly right but with the refactor I have in mind this function will probably go away.